src/inmanta/execute/proxy.py

```diff
@@ -28,14 +28,21 @@
 from inmanta.ast import UnknownException as UnknownException
 from inmanta.ast import UnsetException as UnsetException  # noqa F401
 from inmanta.execute.util import NoneValue, Unknown
+from inmanta.references import Reference
 from inmanta.stable_api import stable_api
 from inmanta.types import PrimitiveTypes
 from inmanta.util import JSONSerializable
 
 if TYPE_CHECKING:
     from inmanta.ast.entity import Entity
+    from inmanta.ast.type import Type as inm_Type
     from inmanta.execute.runtime import Instance, QueueScheduler, Resolver
 
+    TypeResolver = Callable[[type[object]], inm_Type]
+else:
+    # TODO: is there a cleaner way?
+    TypeResolver = ""
```

Comment:
Wouldn't this be better?
```suggestion
    TypeResolver = object
```

Reply:
If you prefer, mypy can't see this and the code doesn't use it....

I updated it

Reply:
This is how we have done it in other places. This was mostly a response to the todo